### ISSUE TITLE
Should GrTags for scalar densities have a Frame template parameter?

### ISSUE THREAD

**Issue by markscheel** [sxs-collaboration/spectre#993]
In #979, the frame and dimension were removed from scalar GrTags.

I have no trouble with this for the lapse, which is the same in all frames.
However, detg in the Inertial frame is different than detg in the Grid frame. 
So should detg have a frame template parameter?

Same for TrK.  I don't think (but need to check) that it is the same in different frames, since it's only a 3-scalar.

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Desired feature:

- [ ] Detail 1
- [ ] Detail 2
- [ ] Detail 3

**Comment by nilsdeppe** [comment]
Having a frame sounds reasonable to me since I expect at least for observing we will have detg in more than one frame in the DataBox.

**Comment by nilsdeppe** [comment]
@kidder

**Comment by kidder** [comment]
yes, this sounds reasonable, but should be done for any non-scalar Scalar
